src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -33,10 +31,22 @@
   private static String DATANODES_KEY = "dataNodes";
   private ZooKeeperState zkState;
 
+  private HashSet<String> unusedNameNodeNames = new HashSet<>();
```

Comment:
Instead of having this here, we should instead use the list of running tasks in LiveState. If you could resolve the TODO storing the NN, JN and DN's separately, then we simply need to loop through the JN and NN list and assign the smallest number that isn't taken. Number of NN's and JN's will be small so this should be quick and we can avoid using the new data structures storing the nameNode names.